backend/src/server/routes/v1/certificate-router.ts

```diff
@@ -62,16 +65,408 @@ const validateDateOrder = (data: { notBefore?: string; notAfter?: string }) => {
   return true;
 };
 
+const validateCertificateRequestFlow = (data: {
+  csr?: string;
+  subjectAlternativeNames?: Array<{ type: ACMESANType; value: string }>;
+  commonName?: string;
+  altNames?: Array<{ type: CertSubjectAlternativeNameType; value: string }>;
+}) => {
+  const hasCSR = !!data.csr;
+  const hasSANs = !!data.subjectAlternativeNames?.length;
+  const hasStandardFields = !!(data.commonName || data.altNames?.length);
+
+  const flowCount = Number(hasCSR) + Number(hasSANs) + Number(hasStandardFields);
+  return flowCount === 1;
+};
+
 export const registerCertificateRouter = async (server: FastifyZodProvider) => {
   server.route({
     method: "POST",
-    url: "/issue-certificate",
+    url: "/",
     config: {
       rateLimit: writeLimit
     },
     schema: {
       hide: false,
       tags: [ApiDocsTags.PkiCertificates],
+      body: z
+        .object({
+          profileId: z.string().uuid(),
+          projectId: z.string().uuid(),
+          commonName: validateTemplateRegexField.optional(),
+          keyUsages: z.nativeEnum(CertKeyUsageType).array().optional(),
+          extendedKeyUsages: z.nativeEnum(CertExtendedKeyUsageType).array().optional(),
+          altNames: z
+            .array(
+              z.object({
+                type: z.nativeEnum(CertSubjectAlternativeNameType),
+                value: z.string().min(1, "SAN value cannot be empty")
+              })
+            )
+            .optional(),
+          signatureAlgorithm: z.nativeEnum(CertSignatureAlgorithm),
+          keyAlgorithm: z.nativeEnum(CertKeyAlgorithm),
```

Comment:
maybe we can unify these under "attributes" property? so it's either CSR is defined or attributes object is defined but not both